### ISSUE TITLE
Feature request: Add workspaces

### ISSUE THREAD

**Issue by notmentaloutlaw** [dino/dino#1313]
Greetings. I have been using dino for a while now and I think its great. It is very user friendly and it runs smoothly. However, there are some things that I would like to suggest to make it even better to allow more people to use this program as well as the xmpp protocol. One being workspaces. Workspaces have been included in other applications such as gajim although gajim has its own issues the workspace feature is something that is highly useful especially if you are on multiple servers. Most matrix clients today use spaces to manage workspaces and a lot of people find that handy. I believe with the inclusion of workspaces it would allow many people to give dino a try. It could even rival the likes of discord given time.

**Comment by gjedeer** [comment]
That's a very confusing feature of Gajim. Gajim was my go-to recommendation for Windows users and after they introduced workspaces I just can't recommend it to non-power-users any more. I myself don't have use case for workspaces too, but maybe I'm just not social enough.

Though I'm not a Dino dev and my opinion is as meaningless as yours.

**Comment by notmentaloutlaw** [comment]
I agree the gajim interface is not that great I understand your point. However, let me describe a common scenario. From a simple UI perspective if a user were to join multiple mucs from multiple servers it can be disorganized on the left hand side. I understand its possible to search and based on the name of the server you can see all the mucs belonging to that server that you previously joined to available. However, I feel that this is primitive compared to other services that many non-power users today are familiar with. For example a lot of people currently use discord and that works how everyone has their server and every room is inside each server and these servers can be managed from the left hand side instead of showing just all the rooms on the left. The same thing goes with matrix clients and workspaces (the user has the option to change it from rooms only to spaces) this clears up a lot of rooms and its easier for users to manage and access rooms with the mouse and gui. I don't know if there is a xmpp protocol that allows servers to combine mucs into one collection of mucs so that users can just click join a specific server and they have access to all the mucs available on the server publically so they don't need to manually type it. But my point isn't that. its a way for users to create sort of bookmarks of collection of rooms on the left hand side instead of having it just rooms. So lets say I joined a few mucs across servers regarding python or something. I can then move all these mucs into this bookmark worskpace and call it python then when i click on it it only shows the rooms that are inside this workspace on the left hand side. It would be great if the user can also manage the position of these rooms in the list to further organise the rooms. I hope this makes sense. I can see a lot of new people in the future (that come from discord/matrix) using the xmpp protocol and so far dino is the most promising in the desktop side. So in order to cater for these new people as well as providing a helpful feature to current users I would recommend this to be at least in the roadmap.

**Comment by licaon-kter** [comment]
@notmentaloutlaw please add some newlines to your wall of text :) :(

**Comment by notmentaloutlaw** [comment]
Sorry about that

**Comment by licaon-kter** [comment]
There are no "discord servers", please try to push against that marketing lie.

Where the rooms are located does not matter, imho. Maybe they are on the same server or not. If they are not public they are not even searchable.

**Comment by licaon-kter** [comment]
Some references for past work on this subject: _(see links in Contributions too)_ https://md.roflcopter.fr/xmpp-spaces#

**Comment by notmentaloutlaw** [comment]
> There are no "discord servers", please try to push against that marketing lie.
> 
> Where the rooms are located does not matter, imho. Maybe they are on the same server or not. If they are not public they are not even searchable.

Yeah I know that I was just using the name that they use for it. And anyway it doesn't matter either way because my point was in terms of the ui itself and how rooms can be combined into one collection.

**Comment by notmentaloutlaw** [comment]
Looking back at this I would think another solution is to create "grouped bookmarks" like instead of having each room just one bookmark I can create a grouped bookmark called physics and it includes physics rooms that i have allocated to it.
